etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationByCode.java

```diff
@@ -0,0 +1,60 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Resource;
+
+public class RemoveObservationByCode implements CustomFhirTransformation {
+    public static final String CODE_NAME = "code";
+    public static final String CODING_SYSTEM_NAME = "codingSystemExtension";
+    public static final String CODING_NAME = "codingExtension";
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, Object> args) {
+        var bundle = (Bundle) resource.getUnderlyingResource();
+        Set<Resource> resourcesToRemove = new HashSet<>();
+
+        for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+            Resource resourceEntry = entry.getResource();
+
+            if (resourceEntry instanceof Observation observation) {
+                removeMatchingObservation(observation, resourcesToRemove, args);
```

Comment:
I think it's better for `removeMatchingObservation` to return the `resourcesToRemove` HashSet instead of passing it as a mutable empty HashSet. It'd reduce unintended side effects